File: Find_The_Original_Array_of_Prefix_Xor.rs
fn main(){
    let v = vec![5,2,0,3,1];
    println!("{:?} ",find_array(v));
}

pub fn find_array(pref:Vec<i32>) ->Vec<i32>{
    let mut v1= vec![];
    v1.push(pref[0]);
    for i in 1..pref.len(){
        v1.push(pref[i] ^ pref[i-1]);
    }
    return v1
}

File: Find_The_Original_Array_of_Prefix_Xor/Find_The_Original_Array_of_Prefix_Xor.go
package main

import "fmt"

func main() {
	pref := []int{5, 2, 0, 3, 1}
	fmt.Println(findArray(pref))
}

func findArray(pref []int) []int {
	res := make([]int, 0, len(pref))
	res = append(res, pref[0])
	for i := 1; i < len(pref); i++ {
		res = append(res, pref[i]^pref[i-1])
	}
	return res
}
